tests/benchmarks/benchmark_gpu_large.py

```diff
@@ -0,0 +1,308 @@
+    exec_dir = os.path.join(cwd, "../..")
+    input_filename = os.path.join(exec_dir, input_path)
+
+    print("Running HELIOTRON fixed rotational transform test.")
+    print("exec_dir=", exec_dir)
+    print("cwd=", cwd)
+
+    args = ["-o", str(desc_h5_path), input_filename, "-vv", "-g"]
+    benchmark(main, args)
+    return None
+
+
+@pytest.mark.slow
+@pytest.mark.benchmark(min_rounds=3, max_time=300, disable_gc=True, warmup=False)
+def test_HELIOTRON_vacuum_run(tmpdir_factory, benchmark):
+    """Benchmark the HELIOTRON vacuum (fixed current) example."""
+    input_path = ".//tests//inputs//HELIOTRON_vacuum"
+    output_dir = tmpdir_factory.mktemp("result")
+    desc_h5_path = output_dir.join("HELIOTRON_vacuum_out.h5")
+    cwd = os.path.dirname(__file__)
+    exec_dir = os.path.join(cwd, "../..")
+    input_filename = os.path.join(exec_dir, input_path)
+
+    print("Running HELIOTRON vacuum (fixed current) test.")
+    print("exec_dir=", exec_dir)
+    print("cwd=", cwd)
+
+    args = ["-o", str(desc_h5_path), input_filename, "-vv", "-g"]
+    benchmark(main, args)
+    return None
+
+
+@pytest.mark.slow
+@pytest.mark.benchmark(min_rounds=3, max_time=300, disable_gc=True, warmup=False)
+def test_ESTELL_run(tmpdir_factory, benchmark):
+    """Benchmark the ESTELL example."""
+    input_path = ".//examples//DESC//ESTELL"
+    output_dir = tmpdir_factory.mktemp("result")
+    desc_h5_path = output_dir.join("ESTELL_out.h5")
+    cwd = os.path.dirname(__file__)
+    exec_dir = os.path.join(cwd, "../..")
+    input_filename = os.path.join(exec_dir, input_path)
+
+    print("Running ESTELL test.")
+    print("exec_dir=", exec_dir)
+    print("cwd=", cwd)
+
+    args = ["-o", str(desc_h5_path), input_filename, "-vv", "-g"]
+    benchmark(main, args)
+    return None
+
+
+@pytest.mark.slow
+@pytest.mark.benchmark(min_rounds=3, max_time=300, disable_gc=True, warmup=False)
+def test_W7X_run(tmpdir_factory, benchmark):
+    """Benchmark the W7X example."""
+    input_path = ".//examples//DESC//W7-X"
+    output_dir = tmpdir_factory.mktemp("result")
+    desc_h5_path = output_dir.join("W7X_out.h5")
+    cwd = os.path.dirname(__file__)
+    exec_dir = os.path.join(cwd, "../..")
+    input_filename = os.path.join(exec_dir, input_path)
+
+    print("Running W7X test.")
+    print("exec_dir=", exec_dir)
+    print("cwd=", cwd)
+
+    args = ["-o", str(desc_h5_path), input_filename, "-vv", "-g"]
+    benchmark(main, args)
+    return None
+
+
+@pytest.mark.slow
+@pytest.mark.benchmark
+def test_perturb_1(SOLOVEV, benchmark):
+    """Benchmark 1st order perturbations."""
+
+    def setup():
+        eq = EquilibriaFamily.load(load_from=str(SOLOVEV["desc_h5_path"]))[-1]
+        objective = get_equilibrium_objective()
+        constraints = get_fixed_boundary_constraints()
+        tr_ratio = [0.01, 0.25, 0.25]
+        dp = np.zeros_like(eq.p_l)
+        objective.build(eq)
+        dp[np.array([0, 2])] = 8e3 * np.array([1, -1])
```

Comment:
should np in this file be replaced with jnp?

Reply:
no it should be fine, it's just used for setting up the problem. The actual compute will still run on the gpu